ament_cmake_target_dependencies/cmake/ament_target_dependencies.cmake

```diff
@@ -65,12 +65,37 @@ function(ament_target_dependencies target)
       if(NOT "${${package_name}_FOUND}")
         message(FATAL_ERROR "ament_target_dependencies() the passed package name '${package_name}' was not found before")
       endif()
-      if(NOT "${${package_name}_INTERFACES}" STREQUAL "")
-        # if a package provides modern CMake interface targets use them
-        # exclusively assuming the classic CMake variables only exist for
-        # backward compatibility
-        list_append_unique(interfaces ${${package_name}_INTERFACES})
-      else()
+      # if a package provides modern CMake interface targets use them
+      # exclusively assuming the classic CMake variables only exist for
+      # backward compatibility
+      set(use_modern_cmake FALSE)
+      if(NOT "${${packag_ename}_TARGETS}" STREQUAL "")
```

Comment:
```suggestion
      if(NOT "${${package_name}_TARGETS}" STREQUAL "")
```

Reply:
Good catch - fixed in d00fe58fa9138a835ae7bdddafa636931e8e2fab.